src/Compilers/CSharp/Portable/Symbols/Source/ExtensionGroupingInfo.cs

```diff
@@ -104,6 +110,37 @@ public Cci.ITypeDefinition GetCorrespondingMarkerType(SourceNamedTypeSymbol type
             throw ExceptionUtilities.Unreachable();
         }
 
+        public Cci.TypeMemberVisibility GetCorrespondingMarkerMethodVisibility(SynthesizedExtensionMarker marker)
+        {
+            Debug.Assert(Cci.TypeMemberVisibility.Public > Cci.TypeMemberVisibility.Assembly);
+            Debug.Assert(Cci.TypeMemberVisibility.Assembly > Cci.TypeMemberVisibility.Private);
+
+            var result = Cci.TypeMemberVisibility.Private;
+
+            foreach (var extension in GetCorrespondingMarkerType((SourceNamedTypeSymbol)marker.ContainingType).UnderlyingExtensions)
+            {
+                foreach (var symbol in extension.GetMembers())
```

Comment:
Why are we using members now to infer the visibility?

Reply:
> Why are we using members now to infer the visibility?

We discussed this in our working group and decided that doing this will be better. For example empty extensions will not have impact on public surface. If all members are internal, there will be no impact on public surface even if extended type is public, etc.